pkg/drivers/host_driver.go

```diff
@@ -0,0 +1,167 @@
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package drivers
+
+import (
+	"encoding/json"
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+
+	"bytes"
+
+	"github.com/GoogleCloudPlatform/container-structure-test/pkg/types/unversioned"
+)
+
+type HostDriver struct {
+	ConfigPath string // path to image metadata config on host fs
+	GlobalVars []unversioned.EnvVar
+}
+
+func NewHostDriver(args DriverConfig) (Driver, error) {
+	return &HostDriver{
+		ConfigPath: args.Metadata,
+	}, nil
+}
+
+func (d *HostDriver) Destroy() {
+	// since we're running on the host, don't do anything
+}
+
+func (d *HostDriver) Setup(envVars []unversioned.EnvVar, fullCommands [][]string) {
+	// since we're running on the host, we'll provide an optional teardown field for`
+	// each test that will allow users to undo the setup they did.
+	// keep track of the original env vars so we can reset later.
+	d.GlobalVars = SetEnvVars(envVars)
+	for _, cmd := range fullCommands {
+		d.ProcessCommand(nil, cmd)
+	}
+}
+
+func (d *HostDriver) Teardown(envVars []unversioned.EnvVar, fullCommands [][]string) {
+	// since we're running on the host, we'll provide an optional teardown field for each test that
+	// will allow users to undo the setup they did.
+	ResetEnvVars(d.GlobalVars)
+	for _, cmd := range fullCommands {
+		d.ProcessCommand(nil, cmd)
+	}
+}
+
+// given a list of environment variable key/value pairs, set these in the current environment.
+// also, keep track of the previous values of these vars to reset after test execution.
+func SetEnvVars(vars []unversioned.EnvVar) []unversioned.EnvVar {
+	var originalVars []unversioned.EnvVar
+	for _, env_var := range vars {
+		originalVars = append(originalVars, unversioned.EnvVar{env_var.Key, os.Getenv(env_var.Key)})
+		if err := os.Setenv(env_var.Key, os.ExpandEnv(env_var.Value)); err != nil {
+			logrus.Fatalf("error setting env var: %s", err)
+		}
+	}
+	return originalVars
+}
+
+func ResetEnvVars(vars []unversioned.EnvVar) {
+	for _, env_var := range vars {
+		var err error
+		if env_var.Value == "" {
+			// if the previous value was empty string, the variable did not
+			// exist in the environment; unset it
+			err = os.Unsetenv(env_var.Key)
+		} else {
+			// otherwise, set it back to its previous value
+			err = os.Setenv(env_var.Key, env_var.Value)
+		}
+		if err != nil {
+			logrus.Fatalf("error resetting env var: %s", err)
+		}
+	}
+}
+
+func (d *HostDriver) ProcessCommand(envVars []unversioned.EnvVar, fullCommand []string) (string, string, int) {
+	originalVars := SetEnvVars(envVars)
+	defer ResetEnvVars(originalVars)
+	cmd := exec.Command(fullCommand[0], fullCommand[1:]...)
```

Comment:
you might want to ensure you don't panic here if fullComand is nil or empty

Reply:
we do arg validation before the tests get run, so if the command is empty we won't get this far